crates/store/re_redap_tests/src/tests/query_dataset.rs

```diff
@@ -0,0 +1,197 @@
+        LayerDefinition::simple("my_partition_id2", &["my/entity"]),
+        LayerDefinition::simple(
+            "my_partition_id3",
+            &["my/entity", "another/one", "yet/another/one"],
+        ),
+    ]);
+
+    let dataset_name = "dataset";
+    service.create_dataset_entry_with_name(dataset_name).await;
+    service
+        .register_with_dataset_name(dataset_name, data_sources_def.to_data_sources())
+        .await;
+
+    let requests = vec![
+        (QueryDatasetRequest::default(), "default"),
+        (
+            QueryDatasetRequest {
+                partition_ids: vec!["my_partition_id3".into()],
+                ..Default::default()
+            },
+            "single_partition",
+        ),
+        (
+            QueryDatasetRequest {
+                entity_paths: vec!["/my/entity".into()],
+                select_all_entity_paths: false,
+                ..Default::default()
+            },
+            "single_entity",
+        ),
+        //TODO(RR-2613): add more test cases here when they are supported by OSS server
+    ];
+
+    for (request, snapshot_name) in requests {
+        query_dataset_snapshot(
+            &service,
+            request,
+            dataset_name,
+            &format!("simple_dataset_{snapshot_name}"),
+        )
+        .await;
+    }
+}
+
+pub async fn query_simple_dataset_with_layers(service: impl RerunCloudService) {
+    let data_sources_def = DataSourcesDefinition::new([
+        LayerDefinition::simple("partition1", &["my/entity"]),
+        LayerDefinition::simple("partition1", &["extra/entity"]).layer_name("extra"),
+        LayerDefinition::simple("partition2", &["another/one"]).layer_name("base"),
+        LayerDefinition::simple("partition2", &["extra/entity"]).layer_name("extra"),
+        LayerDefinition::simple("partition3", &["i/am/alone"]),
+    ]);
+
+    let dataset_name = "dataset_with_layers";
+    service.create_dataset_entry_with_name(dataset_name).await;
+    service
+        .register_with_dataset_name(dataset_name, data_sources_def.to_data_sources())
+        .await;
+
+    query_dataset_snapshot(
+        &service,
+        QueryDatasetRequest::default(),
+        dataset_name,
+        "simple_with_layer",
+    )
+    .await;
+}
+
+/// Test that failure cases return the correct error code.
+pub async fn query_dataset_should_fail(service: impl RerunCloudService) {
+    let dataset_name = "dataset";
+    service.create_dataset_entry_with_name(dataset_name).await;
+
+    let test_cases = vec![
+        (
+            "cannot specify entity paths if `select_all_entity_paths` is true",
+            QueryDatasetRequest {
+                entity_paths: vec!["/entity/path".into()],
+                select_all_entity_paths: true,
+                ..Default::default()
+            },
+            tonic::Code::InvalidArgument,
+        ),
+        //TODO(ab): add more failure cases
```

Comment:
any reason you didn't go ahead here? does this warrant a ticket?

Reply:
So much that needs to be done around here :)

I've started a shopping list here: https://github.com/rerun-io/rerun/issues/11591